### ISSUE TITLE
Set authentication headers?

### ISSUE THREAD

**Issue by fullstackwebdev** [OneGraph/graphiql-explorer#17]
Is it possible to set Authentication headers in OneGraph? Is it missing or hidden? 
Thanks

**Comment by grounzero** [comment]
I'd like to know where this is too.  I fudged it for the moment by adding it directly to the header of the `fetch()` call but this will only last as long as the token.

**Comment by sgrove** [comment]
Copying my answer from [our spectrum](https://spectrum.chat/onegraph/general/can-i-connect-the-onegraph-ide-to-a-local-graphql-endpoint~c701ccda-0e2b-4a94-8b42-e0cdf8192a3b) over here:

Updating the `fetch()` method is exactly the right thing to do right now. There's no public graphical plugin for editing headers, but it'd be another great plugin that should probably ship with the default GraphiQL, as so many people use it.

On [OneGraph's public GraphiQL instance](https://www.onegraph.com/graphiql) our authentication button was built specifically by us to use our authentication mechanism, so it wouldn't do any good to share that.

That said, building the plugin shouldn't be too difficult - if you would like some help, feel free to post some code and we can if we can help. You can see how tools like GraphiQL.app and GraphiQL Workshop have implemented it:

![](https://raw.githubusercontent.com/OlegIlyenko/graphiql-workspace/master/screenshot.png)

Long story short: The explorer needs to work with another (non-existent) plugin to set the headers.

**Comment by sahilrajput03** [comment]
What is the status for this issue, can't we have something for setting authentication header for onegraph client ?

**Comment by sgrove** [comment]
@sahilrajput03 Indeed you can - you have your `fetch` function, don't you?

Feel free to post your code somewhere and I can take a look if you're having trouble!
